appserver/extras/payara-micro/payara-micro-core/src/main/java/fish/payara/micro/impl/UberJarCreator.java

```diff
@@ -241,6 +251,19 @@ public void buildUberJar() {
                     }
                 }
             }
+            
+            if (copyDirectory != null) {
+                String basePath = copyDirectory.getPath().replaceAll(copyDirectory + "$", "");
+                List<File> filesToCopy = fillFiles(copyDirectory);
+                for (File file : filesToCopy) {
+                    
+                        JarEntry deploymentEntry = new JarEntry(copyDirectory + file.getPath().replace(basePath, ""));
+                        jos.putNextEntry(deploymentEntry);
+                        Files.copy(file, jos);
```

Comment:
> @smillidge it's an easy change
I think that the code you linked works only if the original entry is uncompressed, otherwise size != compressed size, throwing an exception as reported in the mailing list. The fix needs to uncompress JAR files if the method is DEFLATED and store them uncompressed, it's not enough to just set the method to STORE.